modin/pandas/base.py

```diff
@@ -222,10 +222,6 @@ def _validate_other(
         self,
         other,
         axis,
-        numeric_only=False,
-        numeric_or_time_only=False,
-        numeric_or_object_only=False,
-        comparison_dtypes_only=False,
```

Comment:
I wonder why type checking was (usually) off before, given I don't see its values set as `True` in most places.
I also wonder how much of a performance hit this type checking introduces...

Reply:
@devin-petersohn thoughts?

Reply:
The check here can (not always) introduce some performance issues, and requires the data types be materialized. That said, we probably need the data types to be materialized for the error checking here.